docs/wagtail-migrations.md

```diff
@@ -0,0 +1,82 @@
+# Wagtail and Django data migrations
```

Comment:
👍  for writing docs for this! When this is live we should drop a link to our approach in [the Wagtail issue](https://github.com/torchbox/wagtail/issues/1101) you linked to.